### PR TITLE
Unit test for serpapi and wikipedia reasoning

### PR DESCRIPTION
**Pull request by Zevleg**

### CODE REVIEW

File: reasoning/build.gradle.kts [hunk 1]
```diff
@@ -34,6 +34,7 @@ detekt {
 
 kotlin {
     jvm {
+        withJava()
```
Comment (raulraja) [inline reasoning/build.gradle.kts:37]: 🤯

File: reasoning/src/commonTest/kotlin/mocks/TestModel.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import com.xebia.functional.tokenizer.ModelType
+import com.xebia.functional.xef.llm.Chat
+import com.xebia.functional.xef.llm.Embeddings
+import com.xebia.functional.xef.llm.models.chat.*
+import com.xebia.functional.xef.llm.models.embeddings.EmbeddingRequest
+import com.xebia.functional.xef.llm.models.embeddings.EmbeddingResult
+import com.xebia.functional.xef.llm.models.usage.Usage
+import kotlinx.coroutines.flow.Flow
+
+class TestModel(override val modelType: ModelType, override val name: String) :
```
Comment (raulraja) [inline reasoning/src/commonTest/kotlin/mocks/TestModel.kt:12]: This is fine, but this testing component can be used in other modules and should be in core in the tests there and consumed here as a library. /cc @javipacheco

File: reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      val model = TestModel(modelType = ModelType.ADA, name = "fake-model")
+
+      val scope = Conversation(LocalVectorStore(TestEmbeddings()), conversationId = conversationId)
+
+      val search = TestSearch(model = model, scope = scope)
+
+      val response = search.invoke("number of Leonardo di Caprio's girlfriends")
+
+      val contentExpected =
+        "Content: Gigi Hadid · Camila Morrone · Camila Morrone · Camila Morrone · Nina Agdal · Rihanna · Rihanna · Kelly Rohrbach."
```
Comment (raulraja) [inline reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt:28]: few names here are repeated; also, we should leave Leonardo in peace and use different examples without having to mention a specific person

File: reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      response shouldBe contentExpected
+    }
+
+    "Tittle should be the same at index 1" {
+      val client = TestSerpApiClient()
+
+      val response =
+        client.search(TestSerpApiClient.SearchData("number of Leonardo di Caprio's girlfriends"))
+
+      val titleExpected = "Leonardo DiCaprio's Dating History: Gisele, Blake Lively, ..."
```
Comment (raulraja) [inline reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt:39]: same here

**Review by raulraja — APPROVED** [review]
thanks @Zevleg !

File: reasoning/src/commonTest/kotlin/mocks/TestModel.kt (hunk 2 above)
Comment (Zevleg) [reply reasoning/src/commonTest/kotlin/mocks/TestModel.kt:12]: How should I put the core dependency in reasoning to import that TestModel I tried this but doesn't work

```kotlin
val commonTest by getting {
            dependencies {
                implementation(libs.kotest.property)
                implementation(libs.kotest.framework)
                implementation(libs.kotest.assertions)
                implementation(libs.mock.engine.test)
                implementation(projects.xefCore)
            }
        }
```

File: reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt (hunk 3 above)
Comment (Zevleg) [reply reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt]: Done

File: reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt (hunk 4 above)
Comment (Zevleg) [reply reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt]: Done

**Review by javipacheco — CHANGES_REQUESTED** [review]
I would like to know your opinion about my comment

File: reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      val scope = Conversation(LocalVectorStore(TestEmbeddings()), conversationId = conversationId)
+
+      val search = TestSearch(model = model, scope = scope)
+
+      val response = search.invoke("german shepper")
+
+      val contentExpected =
+        "Content: The German Shepherd is a breed of working dog developed in Germany from traditional herding and farm dogs. Strongly built and relatively"
+
+      response shouldBe contentExpected
```
[the remark this answers is not in the document]
Comment (Zevleg) [reply reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt:30]: Speaking with Javi, these unit tests are not necessary, everything is mocking, to test serpapi and wikipedia we need to do integration test, I close this PR

**Pull request closed without merging**